node/src/components/storage.rs

```diff
@@ -615,6 +620,15 @@ impl Storage {
                     Some(signatures) => signatures,
                     None => BlockSignatures::new(block_hash, block.header().era_id()),
                 };
+                for (public_key, signature) in signatures.proofs.iter() {
```

Comment:
Isn't there a PR to do this on session start?  Seems like overkill to be validating signatures every time we read a block.

If we really need it here, don't we also need it on the other block and deploy getters?

Reply:
Let's keep this check in even if it is redundant to other checks.

Reply:
Okay, having discussed this we'll get rid of this check.

Fixed in https://github.com/CasperLabs/casper-node/pull/1033/commits/6b316c6752128c2fd6df8bf7bf585d680874f477